src/components/Table.tsx

```diff
@@ -116,7 +158,41 @@ export default class Table extends React.Component<
 		return collection;
 	}
 
-	toggleSort(field: string) {
+	toggleAllChecked() {
+		const allChecked = !this.state.allChecked;
+		const checkedItems = allChecked ? this.props.data.slice() : [];
+
+		if (this.props.onCheck) {
+			this.props.onCheck(checkedItems);
+		}
+
+		this.setState({ allChecked, checkedItems });
+	}
+
+	toggleChecked(item: T) {
+		const rowKey = this.props.rowKey;
+		if (!rowKey) {
+			return false;
+		}
+
+		const identifier = item[rowKey];
+
+		const isChecked = !this.isChecked(item);
+		const checkedItems = isChecked
+			? this.state.checkedItems.concat(item)
+			: reject(this.state.checkedItems, { [rowKey]: identifier });
+
+		if (this.props.onCheck) {
+			this.props.onCheck(checkedItems);
```

Comment:
> Should we also pass `the` item whose checkbox was toggled?

Reply:
@thgreasi I don't think we should. I prefer a behaviour similar to filters, where we simply list the updated object collection (like `setRules`). For most of our use cases, knowing which items was just checked/unchecked is not useful.

Reply:
Ok, I guess that if such a need ever appears, we could add it easily.